server/core/runtime/multienv_step_runner.go

```diff
@@ -48,14 +49,19 @@ func (r *MultiEnvStepRunner) Run(
 		}
 	}
 
-	switch postProcessOutput {
-	case valid.PostProcessRunOutputHide:
-		return "", nil
-	case valid.PostProcessRunOutputShow:
-		return sb.String(), nil
-	default:
-		return sb.String(), nil
+	output := ""
+	for _, processOutput := range postProcessOutput {
+		switch processOutput {
+		case valid.PostProcessRunOutputHide:
+			output = ""
+		case valid.PostProcessRunOutputShow:
+			output = sb.String()
+		default:
+			output = sb.String()
+		}
```

Comment:
If I'm reading this right, it looks like it's saying whatever happens to be the last `PostProcessRunOutputOption` wins? So why even loop over them if we're only going to honor the last one? When similar code is run in `server/core/runtime/run_step_runner.go` it takes the output of one filter and passes it to the next, which I think is what users would expect for a list of `PostProcessRunOutputOption`, can that be implemented here, or at least the difference documented?

Reply:
> So why even loop over them if we're only going to honor the last one?

Reply:
Ahh I see now, yeah that makes sense to me.